src/components/component_base.hpp

```diff
@@ -1,33 +1,18 @@
-#ifndef COMPONENT_BASE_H
```

Comment:
just removing `component_type`, which is now in a separate file. no meaningful changes for review.